packages/headless-react/src/ssr-commerce/commerce-engine.test.tsx

```diff
@@ -0,0 +1,330 @@
+  defineProductList,
+  defineStandaloneSearchBox,
+  NavigatorContextProvider,
+  Product,
+} from '@coveo/headless/ssr-commerce';
+import {act, render, renderHook, screen} from '@testing-library/react';
+import {randomUUID} from 'crypto';
+import {PropsWithChildren} from 'react';
+import React from 'react';
+import {
+  vi,
+  expect,
+  describe,
+  test,
+  beforeEach,
+  MockInstance,
+  afterEach,
+} from 'vitest';
+import {MissingEngineProviderError} from '../errors.js';
+import {defineCommerceEngine} from './commerce-engine.js';
+
+describe('Headless react SSR utils', () => {
+  let errorSpy: MockInstance;
+  const mockedNavigatorContextProvider: NavigatorContextProvider = () => {
+    return {
+      clientId: '123',
+      location: 'https://www.example.com',
+      referrer: 'https://www.google.com',
+      userAgent: 'Mozilla/5.0',
+    };
+  };
+  const sampleConfig = getSampleCommerceEngineConfiguration();
+  beforeEach(() => {
+    errorSpy = vi.spyOn(console, 'error');
+  });
+
+  afterEach(() => {
+    errorSpy.mockClear();
+  });
+
+  test('defines react search engine', () => {
+    const {
+      useEngine,
+      controllers,
+      listingEngineDefinition,
+      searchEngineDefinition,
+      standaloneEngineDefinition,
+      ...rest
+    } = defineCommerceEngine({configuration: sampleConfig});
+    const {
+      fetchStaticState,
+      hydrateStaticState,
+      build,
+      StaticStateProvider,
+      HydratedStateProvider,
+      setNavigatorContextProvider,
+    } = listingEngineDefinition;
+
+    const functions = [
+      fetchStaticState,
+      hydrateStaticState,
+      build,
+      useEngine,
+      StaticStateProvider,
+      HydratedStateProvider,
+      setNavigatorContextProvider,
+    ];
+
+    const engineDefinitions = [
+      searchEngineDefinition,
+      standaloneEngineDefinition,
+      listingEngineDefinition,
+    ];
+
+    functions.forEach((returnValue) =>
+      expect(typeof returnValue).toBe('function')
+    );
+
+    engineDefinitions.forEach((returnValue) =>
+      expect(typeof returnValue).toBe('object')
+    );
+
+    expect(controllers).toEqual({});
+    expect(rest).toEqual({}); // No other return values
+  });
+
+  test('creates a hook based on given controller', () => {
+    const {controllers} = defineCommerceEngine({
+      configuration: sampleConfig,
+      controllers: {
+        productList: defineProductList(),
+        searchBox: defineStandaloneSearchBox({
+          options: {redirectionUrl: '/search'},
+        }),
+      },
+    });
+    expect(typeof controllers.useProductList).toEqual('function');
+    expect(typeof controllers.useSearchBox).toEqual('function');
+  });
+
+  describe('context providers', () => {
+    const productItemTestId = 'product-item';
+    const numProducts = 10;
+    const engineDefinition = defineCommerceEngine({
+      configuration: sampleConfig,
+      // TODO: Generalize tests to test all defined controllers dynamically
```

Comment:
We should mention a Jira in this comment

Reply:
This is from search-engine.test.tsx, we should remove the comment imo